### PR TITLE
refactoring to reduce code size

### PR DESCRIPTION
**Pull request by kuhe**
*Issue #, if available:*

*Description of changes:*

If one or more of the packages in the `/packages` directory has been modified, be sure `yarn changeset add` has been run and its output has
been committed and included in this pull request. See CONTRIBUTING.md.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/core/src/submodules/event-streams/EventStreamSerde.ts [hunk 1]
```diff
@@ -232,14 +232,17 @@ export class EventStreamSerde {
     let explicitPayloadMember = null as null | string;
     let explicitPayloadContentType: undefined | string;
 
-    const isKnownSchema = unionSchema.hasMemberSchema(unionMember);
+    const isKnownSchema = (() => {
+      const struct = unionSchema.getSchema() as StructureSchema;
+      return struct.memberNames.includes(unionMember);
+    })();
```
Comment (kuhe) [inline packages/core/src/submodules/event-streams/EventStreamSerde.ts:238]: This was the only call to hasMemberSchema, so I inlined the function here and removed it from the alpha API.

File: packages/core/src/submodules/event-streams/EventStreamSerde.ts [hunk 2]
```diff
@@ -232,14 +232,17 @@ export class EventStreamSerde {
...
     const additionalHeaders: MessageHeaders = {};
 
     if (!isKnownSchema) {
       // $unknown member
       const [type, value] = event[unionMember];
       eventType = type;
-      serializer.write(SCHEMA.DOCUMENT, value);
+      serializer.write(15 satisfies DocumentSchema, value);
```
Comment (kuhe) [inline packages/core/src/submodules/event-streams/EventStreamSerde.ts:245]: importing even a numeric constant creates a lot of code under certain bundling conditions.

Rather than having a "[magic number](https://en.wikipedia.org/wiki/Magic_number_(programming))" 15, the TS `satisfies` keyword is useful to indicate and type-check the number, allowing usage-finding.

File: packages/core/src/submodules/schema/schemas/NormalizedSchema.ts [hunk 3]
```diff
@@ -328,69 +299,42 @@ export class NormalizedSchema implements INormalizedSchema {
...
+    const sc = this.getSchema();
+    const [isDoc, isMap, isList] = [this.isDocumentSchema(), this.isMapSchema(), this.isListSchema()];
+    const memberSchema =
+      typeof sc === "number"
+        ? 0b0011_1111 & sc
+        : sc && typeof sc === "object" && (isMap || isList)
+          ? ((sc as MapSchema | ListSchema).valueSchema as typeof sc)
+          : isDoc
+            ? (15 satisfies DocumentSchema)
+            : void 0;
```
Comment (kuhe) [inline packages/core/src/submodules/schema/schemas/NormalizedSchema.ts:314]: 🥺 forgive my triple nested ternary please

File: packages/core/src/submodules/serde/schema-serde-lib/schema-date-utils.ts
Comment (kuhe) [inline packages/core/src/submodules/serde/schema-serde-lib/schema-date-utils.ts:1]: This file contains date-parsers that are written differently from the ones used for model-ignorant-codegen but have the same interface.

To save on code size, they do less validation work by trusting the server timestamps. For example, no client-side validation of whether a year contains a leap-day.

File: packages/util-base64/src/constants.browser.ts [hunk 4]
```diff
@@ -1,34 +1,14 @@
...
-alphabetByValue[62] = "+";
-alphabetByEncoding["/"] = 63;
-alphabetByValue[63] = "/";
-
-const bitsPerLetter = 6;
-const bitsPerByte = 8;
-const maxLetterValue = 0b111111;
-
-export { alphabetByEncoding, alphabetByValue, bitsPerLetter, bitsPerByte, maxLetterValue };
+const chars = `ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/`;
```
Comment (kuhe) [inline packages/util-base64/src/constants.browser.ts:1]: 😉

File: packages/core/src/submodules/serde/schema-serde-lib/schema-date-utils.ts
Comment (kuhe) [reply packages/core/src/submodules/serde/schema-serde-lib/schema-date-utils.ts:1]: The same test suite as the regular date parsers is used, except for the aforementioned edge cases.

File: packages/core/src/submodules/schema/schemas/NormalizedSchema.ts [hunk 5]
```diff
@@ -445,44 +388,50 @@ export class NormalizedSchema implements INormalizedSchema {
...
-          return type;
-      }
-    }
-    return "Unknown";
-  }
+/**
+ * @param indicator - numeric indicator for preset trait combination.
+ * @returns equivalent trait object.
+ */
+export function translateTraits(indicator: SchemaTraits): SchemaTraitsObject {
```
Comment (trivikr) [inline packages/core/src/submodules/schema/schemas/NormalizedSchema.ts]: nit: move this function to new file `translateTraits.ts`?
